$OpenBSD: patch-Makefile,v 1.1.1.1 2014/03/15 13:37:18 benoit Exp $
- respect ports CFLAGS
- use OpenBSD shared library naming
- sys.mk defines CFLAGS and LDFLAGS so don't use conditional value assignments
  for those.
--- Makefile.orig	Thu Feb 27 02:05:51 2014
+++ Makefile	Mon Mar 10 12:38:53 2014
@@ -5,20 +5,19 @@ MAKE_DIR     ?= install -d
 INSTALL_DATA ?= install
 
 CC?=gcc
-CFLAGS?=-O2 -Wall -g -D_FORTIFY_SOURCE=2 -fstack-protector -fPIC
-LDFLAGS?=-Wl,-z,now -Wl,-z,relro -Wl,-soname,libscrypt.so.0 -Wl,--version-script=libscrypt.version
+CFLAGS=-Wall -g -D_FORTIFY_SOURCE=2 -fstack-protector -fPIC ${PORTS_CFLAGS}
+LDFLAGS=-Wl,-z,now -Wl,-z,relro -Wl,--version-script=libscrypt.version
 CFLAGS_EXTRA?=-Wl,-rpath=.
 
 all: reference
 
 OBJS= crypto_scrypt-nosse.o sha256.o crypto_scrypt-hexconvert.o crypto-mcf.o b64.o crypto-scrypt-saltgen.o crypto_scrypt-check.o crypto_scrypt-hash.o
 
-libscrypt.so.0: $(OBJS) 
-	$(CC)  $(LDFLAGS) -shared -o libscrypt.so.0 -lc -lm  $(OBJS)
+libscrypt.so.${LIBscrypt_VERSION}: $(OBJS) 
+	$(CC)  $(LDFLAGS) -shared -o libscrypt.so.${LIBscrypt_VERSION} -lc -lm  $(OBJS)
 	ar rcs libscrypt.a  $(OBJS)
 
-reference: libscrypt.so.0 main.o b64.o
-	ln -s -f libscrypt.so.0 libscrypt.so
+reference: libscrypt.so.${LIBscrypt_VERSION} main.o b64.o
 	$(CC) -Wall -o reference main.o  b64.o $(CFLAGS_EXTRA) -L.  -lscrypt
 
 clean:
@@ -33,11 +32,10 @@ devtest:
 	splint crypto-scrypt-saltgen.c +posixlib
 	valgrind ./reference
 
-install: libscrypt.so.0
-	$(MAKE_DIR) $(DESTDIR) $(DESTDIR)$(PREFIX) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INCLUDEDIR)
-	$(INSTALL_DATA) -pm 0755 libscrypt.so.0 $(DESTDIR)$(LIBDIR)
-	cd $(DESTDIR)$(LIBDIR) && ln -s -f libscrypt.so.0 $(DESTDIR)$(LIBDIR)/libscrypt.so
-	$(INSTALL_DATA) -pm 0644 libscrypt.h  $(DESTDIR)$(INCLUDEDIR)
+install: libscrypt.so.${LIBscrypt_VERSION}
+	$(MAKE_DIR) $(LIBDIR) $(INCLUDEDIR)
+	$(INSTALL_DATA) -pm 0755 libscrypt.so.${LIBscrypt_VERSION} $(LIBDIR)
+	$(INSTALL_DATA) -pm 0644 libscrypt.h  $(INCLUDEDIR)
 
 install-static: libscrypt.a
-	$(INSTALL_DATA) -pm 0644 libscrypt.a $(DESTDIR)$(LIBDIR)
+	$(INSTALL_DATA) -pm 0644 libscrypt.a $(LIBDIR)
